pre_processing/utils.py

```
@@ -0,0 +1,127 @@
+        _key (str): residue range ("1-3,5-7")
+    """
+    if len(res_range) == 0:
+        return ""
+    key = str(res_range[0])
+    for i, res in enumerate(res_range):
+        if i == 0:
+            continue
+        elif res_range[i-1] != res - 1:
+            key += f",{res}"
+        if i+1 < len(res_range):
+            if res_range[i+1] != res + 1:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)